### PR TITLE
Add tests for mapSuccess and mapApiError

### PR DESCRIPTION
**Pull request by vhontar**
#22 
Followed by: https://github.com/ioki-mobility/kmp-passenger-api/pull/21

1. Added tests for mapSuccess and mapApiError.
2. Removed `runCatching` and use nullable `ApiErrorBody?`

### CODE REVIEW

File: library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        val fakeHttpClient = createHttpClient(HttpStatusCode.BadGateway, content)
+        val fakeResponse = fakeHttpClient.get("https://127.0.0.1")
+
+        val fakeApiErrorInterceptor = object : ApiErrorInterceptor {
+            override fun intercept(apiErrors: List<ApiErrorBody.ApiError>, httpResponseCode: Int): Boolean {
+                return httpResponseCode == HttpStatusCode.Unauthorized.value
+            }
+        }
+
+        mapApiError(fakeResponse, listOf(fakeApiErrorInterceptor)).let { result ->
```
Comment (StefMa) [inline library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt:79]: Don't use `let` here (and anywhere else 🙃 )
Testing should be divded into three sections:
`given, when, then`

We are testing `mapApiError`. This is part of `when`.
The result should be part of `then`.

**Review by StefMa — COMMENTED** [review]
Thank you!
That looks nice 👏 

Did you already verified that this code works within our project?
Is it possible to parse an nullable `ApiErrorBody` with `failedResponse.body()`? 🤔 

Or is it just tested within tests?

File: library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            assertTrue { result is Result.Error.Api.Intercepted }
+            assertEquals(HttpStatusCode.NotAcceptable.value, result.httpStatusCode)
+        }
+    }
+
+    @Test
+    fun `mapApiError returns unauthorized error`() = runTest {
+        val content = ByteReadChannel.Empty
```
[the remark this answers is not in the document]
Comment (vlh-ioki) [reply library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt:49]: Yes, you can check that by removing `?` from `ApiErrorBody`. It must fail.

File: library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt (hunk 1 above)
Comment (vlh-ioki) [reply library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt:79]: Good, forgot about that!

Comment (vlh-ioki) [reply library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt]: https://github.com/ioki-mobility/kmp-passenger-api/pull/26/commits/fef79ddc2dccc190a8548730a88542b48a2a894b

File: library/src/commonTest/kotlin/com/ioki/passenger/api/Helpers.kt
[the remark this answers is not in the document]
Comment (StefMa) [reply library/src/commonTest/kotlin/com/ioki/passenger/api/Helpers.kt:1]: The file (`Helpers`) should have been renamed 🙃

File: library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import io.ktor.client.request.get
+import io.ktor.http.HttpStatusCode
+import io.ktor.util.InternalAPI
+import io.ktor.utils.io.ByteReadChannel
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+@InternalAPI
```
Comment (StefMa) [inline library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt:14]: Should be renmoved.
This is not an annotation intendeted for clients.

**Review by StefMa — APPROVED** [review]
Super nice!
Thanks for this 👏

**Pull request merged**